src/util.ts

```diff
@@ -15,3 +18,26 @@ export const getSchemaFileName = (cmdId: string): string => {
  */
 export const getKeyNameFromFilename = (file: string): string =>
   file.replaceAll('-', ':').replaceAll('__', '-').replace('.json', '')
+
+export function getAllFiles(dirPath: string, ext: string, allFiles: string[] = []): string[] {
+  let files: string[] = []
+  try {
+    files = fs.readdirSync(dirPath)
```

Comment:
readdirSync has the `withFileTypes` option so you don't have to statSync them all to do isDir.

Can you also make a comment so that when we get everything to node20, we can just use the `recursive` option (it was backported to 18.17)

Reply:
```
export const getAllFiles = (dirPath: string, ext: string, allFiles: string[] = []): string[] =>
  safeReadDirSync(dirPath)
    .flatMap((f) =>
      f.isDirectory() ? getAllFiles(path.join(dirPath, f.name), ext, allFiles) : path.join(dirPath, f.name),
    )
    .filter((f) => f.endsWith(ext))

const safeReadDirSync = (dirPath: string): fs.Dirent[] => {
  try {
    return fs.readdirSync(dirPath, {withFileTypes: true})
  } catch {
    return []
  }
}
```